README.md

```diff
@@ -24,21 +23,129 @@ Join us to help define the direction and implementation of this project!
 
+## Installation Instructions
 
+WARNING: It is not recommended to install Gatekeeper on a production cluster. The project is in alpha and may or may not uninstall cleanly.
+
+### Installation
+
+Currently the most reliable way of installing Gatekeeper is to build and install from HEAD:
+
+   * Make sure [Kubebuilder is installed](https://book.kubebuilder.io/getting_started/installation_and_setup.html)
+   * Clone the Gatekeeper repo to your local system
+   * Make sure you have a container registry you can write to that is readable by the target cluster
+   * cd to the repository directory
+   * run `make docker-build REPOSITORY=<YOUR DESIRED DESTINATION DOCKER IMAGE>`
+   * run `make docker-push-release REPOSITORY=<YOUR DESIRED DESTINATION DOCKER IMAGE>`
+   * make sure your kubectl context is set to the desired installation cluster
+   * run `make deploy`
+
+### Uninstallation
+
+Note that this is not a clean uninstall. There may be some CRDs that are not cleaned up
+
+   * cd to the repository directory
+   * run `make uninstall`
+
+## How to Use Gatekeeper
+
+### Constraint Templates
+
+Before you can define a constraint, you must first define a `ConstraintTemplate`, which describes both the Rego that enforces the constraint and the schema of the constraint. The schema of the constraint allows an admin to fine-tune the behavior of a constraint, much like arguments to a function.
+
+Here is an example constraint template that requires all labels described by the constraint to be present:
+
+```yaml
+apiVersion: templates.gatekeeper.sh/v1alpha1
+kind: ConstraintTemplate
+metadata:
+  name: k8srequiredlabels
+spec:
+  crd:
+    spec:
+      names:
+        kind: K8sRequiredLabels
+        listKind: K8sRequiredLabelsList
+        plural: k8srequiredlabels
+        singular: k8srequiredlabels
+      validation:
+        # Schema for the `parameters` field
+        openAPIV3Schema:
+          properties:
+            labels:
+              type: array
+              items: string
+  targets:
+    - target: admission.k8s.gatekeeper.sh
+      rego: |
+        package k8srequiredlabels
+
+        deny[{"msg": msg, "details": {"missing_labels": missing}}] {
+          provided := {label | input.review.object.metadata.labels[label]}
+          required := {label | label := input.constraint.spec.parameters.labels[_]}
+          missing := required - provided
+          count(missing) > 0
+          msg := sprintf("you must provide labels: %v", [missing])
+        }
+```
+
+### Constraints
+
+Constraints are then used to inform Gatekeeper that the admin wants a ConstraintTemplate to be enforced, and how. This constraint uses the `K8sRequiredLabels` constraint above to make sure the `gatekeeper` label is defined on all namespaces:
```

Comment:
nit: uses the `K8sRequiredLabels` constraint template above

Reply:
Good catch.